Packs/HelloWorld/Integrations/FeedHelloWorld/FeedHelloWorld.py

```diff
@@ -0,0 +1,437 @@
+
+
+Entry Point
+-----------
+
+This is the integration code entry point. It checks whether the ``__name__``
+variable is ``__main__`` , ``__builtin__`` (for Python 2) or ``builtins`` (for
+Python 3) and then calls the ``main()`` function. Just keep this convention.
+
+"""
+
+from typing import Dict, List, Optional
+
+import urllib3
+
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+# disable insecure warnings
+urllib3.disable_warnings()
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    This Client implements API calls, and does not contain any Demisto logic.
+    Should only do requests and return data.
+    It inherits from BaseClient defined in CommonServer Python.
+    Most calls use _http_request() that handles proxy, SSL verification, etc.
+    For this HelloWorld Feed implementation, no special attributes defined
+    """
+
+    def build_iterator(self) -> List:
+        """Retrieves all entries from the feed.
+        Returns:
+            A list of objects, containing the indicators.
+        """
+
+        res = self._http_request('GET',
+                                 url_suffix='',
+                                 full_url=self._base_url,
+                                 resp_type='text',
+                                 )
+
+        result = []
+
+        # In this case the feed output is in text format, so extracting the indicators from the response requires
+        # iterating over it's lines solely. Other feeds could be in other kinds of formats (CSV, MISP, etc.), or might
+        # require additional processing as well.
+        try:
+            indicators = res.split('\n')
+
+            for indicator in indicators:
+                if auto_detect_indicator_type(indicator):
+                    result.append({
+                        'value': indicator,
+                        'type': auto_detect_indicator_type(indicator),
+                        'FeedURL': self._base_url
+                    })
+
+        except ValueError as err:
+            demisto.debug(str(err))
+            raise ValueError(f'Could not parse returned data to Json. \n\nError massage: {err}')
+        return result
+
+
+def test_module(client: Client) -> str:
+    """Builds the iterator to check that the feed is accessible.
+    Args:
+        client: Client object.
+    Returns:
+        Outputs.
+    """
+    # INTEGRATION FEED DEVELOPER TIP
+    # Client class should raise the exceptions, but if the test fails
+    # the exception text is printed to the Cortex XSOAR UI.
+    # If you have some specific errors you want to capture (i.e. auth failure)
+    # you should catch the exception here and return a string with a more
+    # readable output (for example return 'Authentication Error, API Key
+    # invalid').
+    # Cortex XSOAR will print everything you return different than 'ok' as
+    # an error
+
+    client.build_iterator()
```

Comment:
Does this provide a clear error message when the base_url is invalid? It should be an example where capturing a generic error (i.e. an HTTP 404) is captured and shown as a user error that is clear to understand.

Reply:
Yes, the error is being raised from '_http_request' function in CSP and is being caught by the main function (that modifies the error message to imply the error reason).